spec/system/page_title_spec.rb

```diff
@@ -0,0 +1,97 @@
+        user = create :user
+        user.lock_access!
+        visit user_unlock_path(unlock_token: user.reload.unlock_token)
+      end
+    end
+  end
+
+  context 'when user is logged in' do
+    include_context 'with user'
+
+    it { has_title('Home page') { visit root_path } }
+    it { has_title('My learning') { visit my_learning_path } }
+    it { has_title('About training') { visit course_overview_path } }
+    it { has_title('User timeout') { visit users_timeout_path } }
+    it { has_title('Cookie policy') { visit setting_path(id: :cookie_policy) } }
+    it { has_title('Change your password') { visit edit_user_registration_path } }
+    it { has_title('Resend your confirmation') { visit new_user_confirmation_path } }
+    it { has_title('About you') { visit extra_registrations_path } } # redirects to first step of extra registration, which is name
+    it { has_title('About you') { visit edit_extra_registration_path(:name) } }
+    it { has_title('About your setting') { visit edit_extra_registration_path(:setting) } }
+    it { has_title('Change name') { visit edit_name_user_path } }
+    it { has_title('Change email address') { visit edit_email_user_path } }
+    it { has_title('Change password') { visit edit_password_user_path } }
+    it { has_title("Change your setting's postcode information") { visit edit_postcode_user_path } }
+    it { has_title("Change your setting's Ofsted number") { visit edit_ofsted_number_user_path } }
+    it { has_title('Check email confirmation') { visit check_email_confirmation_user_path } }
+    it { has_title('Check email password reset') { visit check_email_password_reset_user_path } }
+    it { has_title('My account') { visit user_path } }
+
+    context 'and using training module' do
+      let(:training_module_id) { 'alpha' }
+
+      it { has_title('First Training Module') { visit training_module_content_pages_path(training_module_id: training_module_id) } }
+      it { has_title('First Training Module') { visit training_module_content_page_path(training_module_id: training_module_id, id: 'before-you-start') } }
+      # TODO: How to set up questionnaire?
+      # it { has_title('First Training Module') { visit training_module_questionnaire_path(training_module_id: training_module_id, id: 'before-you-start') } }
+      it { has_title('First Training Module') { visit training_module_formative_assessment_path(training_module_id: training_module_id, id: '1-1-4') } }
+      it { has_title('First Training Module') { visit training_module_confidence_check_path(training_module_id: training_module_id, id: '1-3-3-1') } }
+      it { has_title('Assessments results') { visit training_module_assessments_result_path(training_module_id: training_module_id, id: '1-3-2-5') } }
+      it { has_title('End of module test') { visit training_module_retake_quiz_path(training_module_id: training_module_id) } }
+
+      it {
+        ahoy = Ahoy::Tracker.new(user: user, controller: 'content_pages')
```

Comment:
There are helper methods in the `with progress` shared context that may be useful. Like `view_module_page_event`

Reply:
Or `view_whole_module`

Reply:
It doesn't work with system tests (user needs to be logged in) and it seems easier to do it this way.

Reply:
It does you just add `include_context 'with user'`

Reply:
Without it the GovUkComponents banner was the issue, not the matcher